lib/structures/Invite.js

```diff
@@ -29,21 +30,9 @@ class Invite extends Base {
     constructor(data, client) {
         super();
         this._client = client;
+        this.id = data.code;
         this.code = data.code;
         this.channel = data.channel;
-        if(data.guild) {
```

Comment:
Hides information in invites from REST from guilds that aren't cached

Reply:
Could you explain this to me? I don't understand how having this inside an if is useful.

```js
if (data.x) {
  this.guild = y
}
```
is equivalent to:
```js
this.guild = data.x ? y : undefined
```

Let's suppose the goal of this if was to hide info in invites on REST calls from UNCACHED guilds. If `data.guild` is not passed and the user does `invite.guild` they will get `undefined`. The same is true with a ternary. If a `data.guild` is not passed it will be undefined.

Am I missing something here?

Reply:
![image](https://user-images.githubusercontent.com/23035000/72683637-1b49eb00-3aa7-11ea-94a4-693731035440.png)


![image](https://user-images.githubusercontent.com/23035000/72683648-24d35300-3aa7-11ea-91ee-769650e002b4.png)


![image](https://user-images.githubusercontent.com/23035000/72683656-2ac93400-3aa7-11ea-9545-a5432aaa4391.png)


![image](https://user-images.githubusercontent.com/23035000/72683660-2ef55180-3aa7-11ea-85d1-c251301671b0.png)


![image](https://user-images.githubusercontent.com/23035000/72683662-33ba0580-3aa7-11ea-9234-39cb33a32d42.png)
---
Bot is NOT on this guild:
![image](https://user-images.githubusercontent.com/23035000/72683688-7085fc80-3aa7-11ea-9a1f-77078f8b5a49.png)

![image](https://user-images.githubusercontent.com/23035000/72683695-7aa7fb00-3aa7-11ea-9ce7-a21e86a0d021.png)



If there is any way that i did not fetch an invite let me know and id be happy to test that way as well

Reply:
You replaced this block with `this.guild = data.guild ? client.guilds.get(data.guild.id) : undefined;`. If the guild is cached, that's fine. If the guild isn't cached, the new code would result in `undefined`. The old code would result in a partial guild-like object, which is better than `undefined`. Useful especially for invites from REST.

Also, having to check if it's a guild-like object or `Guild` instance is annoying. I'd prefer it to stay as-is.

Reply:
Done! Removed changes to this file.